File: Solutions/1-TwoSum/TwoSum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val hashMap = HashMap<Int, Int>() // {compliment, index}

        for (i in nums.indices) {
            val compliment = target - nums[i]

            if (hashMap.containsKey(compliment)) {
                return intArrayOf(hashMap[compliment]!!, i)
            } else {
                hashMap[nums[i]] = i
            }
        }
        return intArrayOf()
    }
}

File: Solutions/1-TwoSum/TwoSum.go
package solutions

func twoSum(nums []int, target int) []int {
	m := make(map[int]int)

	for currIndex, num := range nums {
		compliment := target - num

		if complimentIndex, ok := m[compliment]; ok {
			return []int{currIndex, complimentIndex}
		} else {
			m[num] = currIndex
		}
	}

	return []int{-1, -1}
}
